doc/api/esmvalcore.api.recipe_output.rst

```diff
@@ -0,0 +1,138 @@
+      DataFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.nc')
+      ImageFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.png')
+      ImageFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.png')
+
+
+Output is grouped by the task that produced them. They can be accessed like
+a dictionary.
+
+.. code:: python
+
+    >>> task_output = recipe_output['map/script1']
+    >>> task_output
+    map/script1:
+      DataFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.nc')
+      DataFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.nc')
+      ImageFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.png')
+      ImageFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.png')
+
+
+The task output has a list of files associated with them, usually image
+(``.png``) or data files (``.nc``). To get a list of all files, use
+``.files``:
+
+.. code:: python
+
+    >>> print(task_output.files)
+    (DataFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.nc'),
+    ..., ImageFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.png'))
+
+
+It is also possible to select the image files or data files only:
+
+.. code:: python
+
+    >>> for image_file in task_output.image_files:
+    >>>     print(image_file)
+    ImageFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.png')
+    ImageFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.png')
+
+    >>> for data_file in task_output.data_files:
+    >>>     print(data_file)
+    DataFile('CMIP5_CanESM2_Amon_historical_r1i1p1_tas_2000-2000.nc')
+    DataFile('CMIP6_BCC-ESM1_Amon_historical_r1i1p1f1_tas_2000-2000.nc')
+
+
+
+Working with image files
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Output comes in two kinds, ``DataFile``s, which correspond to data
+files in ``.nc`` format and ``ImageFile``s which correspond to plots
+in ``.png`` format. Both object are derived from the same base class
+(``OutputFile``) and therefore share most of the functionality.
+
+Image output uses IPython magic to plot themselves in a notebook
+environment.
+
+.. code:: python
+
+    >>> image_file = recipe_output['map/script1'].image_files[0]
+    >>> image_file
+
+For example:
+
+.. image:: /figures/api_recipe_output.png
+   :width: 600
+
+Using :py:mod:`IPython.display`, it is possible to show all image files.
+
+.. code:: python
+
+    >>> from IPython.display import display
+    >>>
+    >>> task = recipe_output['map/script1']
+    >>> for image_file in task.image_files:
+    >>>      display(image_file)
+
+
+Working with data output
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Data files can be easily loaded using ``xarray``:
+
+.. code:: python
+
+    >>> data_file = recipe_output['timeseries/script1'].data_files[0]
+    >>> data = data_file.load_xarray()
+    >>> type(data)
+    xarray.core.dataset.Dataset
+
+
+Or ``iris``:
+
+.. code:: python
+
+    >>> cube = data_file.load_iris()
+    >>> type(cube)
+    iris.cube.CubeList
+
+
+API reference
```

Comment:
This page is so much better already! I just wonder if we should still add something about citation info or dataset attributes, etc. Since these attributes are listed in the API reference below, it might be nice to provide a very short example to show how they can be used.

And should we include a statement about the output of the preprocessing tasks? Since they are removed unless `remove_preproc_dir` is set to false?

Reply:
> This page is so much better already! I just wonder if we should still add something about citation info or dataset attributes, etc. Since these attributes are listed in the API reference below, it might be nice to provide a very short example to show how they can be used.

I will add a few lines about this. 👍

> And should we include a statement about the output of the preprocessing tasks? Since they are removed unless `remove_preproc_dir` is set to false? (PS: I just realized the whole preprocessor file parsing still needs to be added. Nevertheless, a statement about this might be helpful.)

Good point! I will look into how handle preprocessor output / specific task output / edge cases in a separate PR (including documentation) and keep this PR limited to recipe output only.